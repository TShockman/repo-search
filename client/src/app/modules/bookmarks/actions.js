export const NAME = "bookmarks";

export const ADD_BOOKMARK = `${NAME}/ADD`;
export const GET_BOOKMARKS_REQUESTED = `${NAME}/GET_REQUESTED`;
export const GET_BOOKMARKS_FULFILLED = `${NAME}/GET_FULFILLED`;