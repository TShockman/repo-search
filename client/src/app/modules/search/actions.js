export const NAME = 'search';

export const UPDATE_QUERY = `${NAME}/UPDATE_QUERY`;
export const SEARCH_REQUESTED = `${NAME}/SEARCH_REQUESTED`;
export const SEARCH_FULFILLED = `${NAME}/SEARCH_FULFILLED`;
