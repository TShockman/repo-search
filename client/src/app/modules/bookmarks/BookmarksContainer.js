import {connect} from 'react-redux';
import Bookmarks from './Bookmarks';
import {GET_BOOKMARKS_REQUESTED} from './actions';
import {selectBookmarksState} from './selectors';

function mapStateToProps(state) {
  const {repos} = selectBookmarksState(state);
  return {repos};
}

function mapDispatchToProps(dispatch) {
  return {
    getBookmarks: () => {
      return dispatch({
        type: GET_BOOKMARKS_REQUESTED
      })
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Bookmarks);