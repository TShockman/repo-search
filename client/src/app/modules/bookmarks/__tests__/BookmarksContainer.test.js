jest.mock('../Bookmarks', () => 'Bookmarks');

import React from 'react';
import renderer from 'react-test-renderer';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import bookmarksReducer, {BookmarkState} from '../reducer';
import {NAME} from '../actions';
import BookmarksContainer from '../index';

describe('BookmarksContainer', () => {
  it('Provides Bookmarks with correct props', () => {
    const store = createStore(bookmarksReducer, {
      [NAME]: new BookmarkState({
        repos: [
          {id: 123, full_name: "OneTwo/Three"},
          {id: 456, full_name: "FourFive/Six"},
          {id: 789, full_name: "SevenEight/Nine"}
        ]
      })
    });
    const tree = renderer.create(
        <Provider store={store}>
          <BookmarksContainer/>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});