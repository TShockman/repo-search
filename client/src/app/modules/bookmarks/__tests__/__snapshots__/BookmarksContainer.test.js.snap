// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarksContainer Provides Bookmarks with correct props 1`] = `
<Bookmarks
  getBookmarks={[Function]}
  repos={
    Array [
      Object {
        "full_name": "OneTwo/Three",
        "id": 123,
      },
      Object {
        "full_name": "FourFive/Six",
        "id": 456,
      },
      Object {
        "full_name": "SevenEight/Nine",
        "id": 789,
      },
    ]
  }
/>
`;
